ros/angel_system_nodes/angel_system_nodes/task_monitor.py

```diff
@@ -117,11 +147,23 @@ def listener_callback(self, activity_msg):
             # Update state tracking vars
             self._previous_step = self._current_step
             self._current_step = self._task.state
+
         except transitions.core.MachineError as e:
             pass
 
         self.publish_task_state_message()
 
+        # Check to see if this new state has a timer associated with it
+        try:
+            if self._task.machine.states[self._task.state].timer_length > 0:
+                # Spawn a thread to track the timer and publish state messages
+                t = threading.Thread(target=self.task_timer_thread)
+                t.start()
+                self._timer_active = True
```

Comment:
The use of a lock will mitigate this, but we probably want to set the `_timer_active` variable *before* starting the thread (which before could theoretically set `_timer_active` before the main thread gets to the next line).
```suggestion
                with self._timer_lock:
                    self._timer_active = True
                    t = threading.Thread(target=self.task_timer_thread)
                    t.start()
```